services/tasks/src/test/java/tech/pegasys/pantheon/services/tasks/FlatFileTaskCollectionTest.java

```diff
@@ -72,8 +69,7 @@ public void shouldRollFilesWhenSizeExceeded() throws Exception {
         removedTasks.add(queue.remove().getData());
       }
 
-      // Fully read files should have been removed.
-      assertThat(getCurrentFiles(dataDir)).doesNotContain(firstFile);
+      assertThat(queue.getReadFileNumber()).isEqualTo(0);
```

Comment:
If the first file has been read and closed, shouldn't the readFile number increment to 1??
```suggestion
      assertThat(queue.getReadFileNumber()).isEqualTo(1);
```

Reply:
The file number doesn't get updated until it actually reads the file.  This change will break the test but I'm adding a tweak to make sure it advances.